src/imp/tracer_imp.js

```diff
@@ -183,24 +184,57 @@ export default class TracerImp extends EventEmitter {
     }
 
     startSpan(fields) {
-        let spanImp = new SpanImp(this);
+        // First, assemble the SpanContextImp for our SpanImp.
+        //
+        // Normalize `fields.reference` and `fields.references` (at most one
+        // will be defined).
+        let refs = fields.references;
+        if (fields.reference) {
+            refs = [fields.reference];
+        }
+        let parCtxImp = null;
+        if (refs) {
+            for (let i = 0; i < refs.length; i++) {
+                let ref = refs[i];
+                if (ref.type() === this._interface.REFERENCE_CHILD_OF ||
+                        ref.type() === this._interface.REFERENCE_FOLLOWS_FROM) {
+                    parCtxImp = ref.spanContext().imp();
+                    break;
+                }
+            }
+        }
+        let newCtx = null;
+        if (parCtxImp === null) {
+            // Root SpanContext case:
+            newCtx = new SpanContextImp(
+                    this._platform.generateUUID(),
+                    this.generateTraceGUIDForRootSpan());
+        } else {
+            // Child SpanContextImp case:
+            newCtx = new SpanContextImp(
+                    this._platform.generateUUID(),
+                    parCtxImp._traceGUID);
+        }
```

Comment:
nit, but this might read a little easier as:

```
let traceGUID = parCtxImp ? parCtxImp._traceGUID : this.generateTraceGUIDForRootSpan();
let newCtx = new SpanContextImp(this._platform.generateUUID(), traceGUID);
```